jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -512,10 +533,22 @@ private void checkClassAccess(Class<?> targetClass) throws IllegalAccessExceptio
 					}
 				}
 			}
-			
+
 			/*[MSG "K0587", "'{0}' no access to: '{1}'"]*/
 			throw new IllegalAccessException(com.ibm.oti.util.Msg.getString("K0587", accessClass.getName(), targetClass.getName()));  //$NON-NLS-1$
 		}
+
+		void checkLookupVisibility(Class<?> targetClass) throws IllegalAccessException, IllegalAccessError {
+			if (this.equals(PUBLIC_LOOKUP)) {
```

Comment:
Rather than checking is identically PUBLIC_LOOKUP, we should check for the `UNCONDITIONAL` mode access bit - see https://docs.oracle.com/javase/9/docs/api/java/lang/invoke/MethodHandles.html#publicLookup--